### ISSUE TITLE
Augment with Keypoints sometimes not work

### ISSUE THREAD

**Issue by konstantin-frolov** [albumentations-team/albumentations#703]
## 🐛 Bug

Sometimes Keypoint Augmentation not working for augment data on the fly with custom Keras datagenerator.
I try create augmentation "on the fly" with custom keras datagenerator. Class CustomSeq get images and keypoints(x, y coordinates) from pandas DataFrame and augmented it.

## To Reproduce

1. Download [csv](https://drive.google.com/file/d/1fPD2pAM2u5IbfnDpBFeeeJlbxufV9ySO) and two images: [one](https://drive.google.com/file/d/1zSFfKDisvWSGwKPDd41bvcqaiUypjX24), [two](https://drive.google.com/file/d/1EU6isFYmj-qBWseb967dCkpnjj8Dii2_)
2. put csv and 2 images to some dir
3. create CustomSeq datagen for Keras
4. get item from datagen with method __getitem__(idx)
5. visualize Image and Keypoints with pyplot
6. repeat 4-5 N times

Sometimes the augmentation works fine. But sometimes the image is augmented, but the keypoints are not.
Keypoints are augmented for only one transformation (HorizontalFlip, VerticalFlip, or RandomRotateScale). But the image can be transformed at the same time using HorizontalFlip and VerticalFlip.

```python
import numpy as np
import imageio
import os
import matplotlib.pyplot as plt
import pandas as pd
import albumentations as A
import cv2
import json
from tensorflow.python.keras.utils.data_utils import Sequence

def extract_coordinates(df):
    full_coordinates = df['region_shape_attributes']
    ls_coordinates = []
    for coordinates in full_coordinates:
        coordinates = json.loads(coordinates)
        ls_coordinates.append([coordinates['cx'], coordinates['cy']])
    return np.array(ls_coordinates, dtype=np.float32)

def rescale_image(image):
    return (image / np.max(image) * 255.).astype(np.float32)

class CustomSeq(Sequence):
    def __init__(self, path2imgs, df, batch_size, augmentations=None, mode='train'):
        self.path2imgs = path2imgs
        self.df = df
        self.img_list = self.df['filename']
        self.y = extract_coordinates(self.df)
        self.batch_size = batch_size
        self.augmentations = augmentations
        self.mode = mode.lower()
        
    def __len__(self):
        return int(np.ceil(len(self.df) / float(self.batch_size)))
    
    def on_epoch_end(self):
        self.indexes = range(len(self.img_list))
        if self.mode == 'train':
            self.indexes = random.sample(self.indexes, k=len(self.indexes))
    
    def get_batch_labels(self, idx, shapes):
        y_batch = self.y[idx * self.batch_size: (idx+1) * self.batch_size]
        return y_batch
    
    def get_batch_images(self, idx):
        x_batch = []
        shapes = []
        img_names = self.img_list[idx * self.batch_size: (idx+1) * self.batch_size]
        for img_name in img_names:
            image = imageio.imread(os.path.join(self.path2imgs, img_name))
            image = rescale_image(image)
            x_batch.append(image)
            shapes.append(image.shape)
        return x_batch, np.array(shapes)
    
    def __getitem__(self, idx):
        x_batch, shapes = self.get_batch_images(idx)
        y_batch = self.get_batch_labels(idx, shapes)
        if self.augmentations:
            # walk around images and keypoints
            for i, (x_item, y_item) in enumerate(zip(x_batch, y_batch)):
                transformed = self.augmentations(image=x_item, keypoints=np.expand_dims(y_item, axis=0))
                # Rewrite image and keypoints values in not augmented batch
                x_batch[i], y_batch[i] = transformed['image'], transformed['keypoints'][0]
        return x_batch, y_batch


transform = A.Compose([
    A.HorizontalFlip(p=0.5),
    A.VerticalFlip(p=0.5),
    #A.InvertImg(p=0.5),
    #A.ShiftScaleRotate(shift_limit=0, scale_limit=0.1, rotate_limit=15, p=0.5),
    A.ToFloat(max_value=255.)
], keypoint_params=A.KeypointParams(format='xy'))

path2png_imgs = os.getcwd()
df = pd.read_csv('vgg_annotate_crop.csv', header=0)

data = CustomSeq(path2png_imgs, df, 1, augmentations=transform)

images, point = data.__getitem__(0)

images[0] = cv2.circle(images[0], list(map(tuple, point.astype(np.int).tolist()))[0], 30, (1), -1)
plt.figure(figsize=(10, 10))
plt.imshow(images[0], cmap='gray')
```

## Environment

 - Albumentations version (e.g., 0.4.6):
 - Python version (e.g., 3.7):
 - Os: Win10
 - How you installed albumentations (`conda`, `pip`, source): conda
 - Any other relevant information:

**Comment by innat** [comment]
Please, make it public to download the samples.

**Comment by konstantin-frolov** [comment]
> Please, make it public to download the samples.

Sorry. I fix it.

**Comment by Dipet** [comment]
I can not reproduce problem.
Please try to determine random seed when the error occures
```python
init_seed = 0
for i in range(10):
    seed = i + init_seed
    random.seed(seed)
    np.random.seed(seed)

    transform = A.Compose([
        A.HorizontalFlip(p=0.5),
        A.VerticalFlip(p=0.5),
        # A.InvertImg(p=0.5),
        # A.ShiftScaleRotate(shift_limit=0, scale_limit=0.1, rotate_limit=15, p=0.5),
        A.ToFloat(max_value=255.)
    ], keypoint_params=A.KeypointParams(format='xy'))

    path2png_imgs = os.getcwd()
    df = pd.read_csv('/home/druzhinin/HDD/NewDownloads/vgg_annotate_crop.csv', header=0)

    data = CustomSeq(path2png_imgs, df, 1, augmentations=transform)

    images, point = data.__getitem__(0)
    images = [cv2.cvtColor(i, cv2.COLOR_GRAY2RGB) for i in images]

    images[0] = cv2.circle(images[0], list(map(tuple, point.astype(np.int).tolist()))[0], 30, (1, 0, 0), -1)
    plt.figure(figsize=(10, 10))
    plt.imshow(images[0], cmap='gray')
    plt.show()
```

**Comment by konstantin-frolov** [comment]
> I can not reproduce problem.
> Please try to determine random seed when the error occures
It's really strange. I ran the code in Pycharm with your loop to debug the code. And it works fine. 
But in Jupyter Notebook problem still exists. It looks like this is a problem with Jupyter.
Thank you for support!
